tests/unittests/test_datasource/test_azure_helper.py

```diff
@@ -496,73 +513,174 @@ def setUp(self):
         reporter = azure_helper.GoalStateHealthReporter(
             azure_helper.GoalState(mock.MagicMock(), mock.MagicMock()),
             azure_helper.AzureEndpointHttpClient(mock.MagicMock()),
-            self.test_endpoint)
+            self.test_azure_endpoint)
         generated_health_document = reporter.build_report(
             incarnation=self.default_parameters['incarnation'],
             container_id=self.default_parameters['container_id'],
             instance_id=self.default_parameters['instance_id'],
             status=self.provisioning_success_status)
+
         self.assertEqual(health_document, generated_health_document)
-        self.assertIn(
-            '<GoalStateIncarnation>{}</GoalStateIncarnation>'.format(
-                str(self.default_parameters['incarnation'])),
-            generated_health_document)
-        self.assertIn(
-            ''.join([
-                '<ContainerId>',
-                self.default_parameters['container_id'],
-                '</ContainerId>']),
-            generated_health_document)
-        self.assertIn(
-            ''.join([
-                '<InstanceId>',
-                self.default_parameters['instance_id'],
-                '</InstanceId>']),
-            generated_health_document)
-        self.assertIn(
-            ''.join([
-                '<State>',
-                self.provisioning_success_status,
-                '</State>']),
-            generated_health_document
+
+        generated_xroot = self._xroot_from_raw_xml(generated_health_document)
+        self.assertEqual(
+            self._text_from_xpath_in_xroot(
+                generated_xroot, './GoalStateIncarnation'),
+            str(self.default_parameters['incarnation']))
+        self.assertEqual(
+            self._text_from_xpath_in_xroot(
+                generated_xroot, './Container/ContainerId'),
+            str(self.default_parameters['container_id']))
+        self.assertEqual(
+            self._text_from_xpath_in_xroot(
+                generated_xroot,
+                './Container/RoleInstanceList/Role/InstanceId'),
+            str(self.default_parameters['instance_id']))
+        self.assertEqual(
+            self._text_from_xpath_in_xroot(
+                generated_xroot,
+                './Container/RoleInstanceList/Role/Health/State'),
+            escape(self.provisioning_success_status))
+        self.assertIsNone(
+            self._text_from_xpath_in_xroot(
+                generated_xroot,
+                './Container/RoleInstanceList/Role/Health/Details'))
+        self.assertIsNone(
+            self._text_from_xpath_in_xroot(
+                generated_xroot,
+                './Container/RoleInstanceList/Role/Health/Details/SubStatus'))
+        self.assertIsNone(
+            self._text_from_xpath_in_xroot(
+                generated_xroot,
+                './Container/RoleInstanceList/Role/Health/Details/Description')
         )
```

Comment:
Improved way of testing the health report XML docs. Using `_text_from_xpath_in_xroot` ensures that the XML is valid (because `_text_from_xpath_in_xroot` raises exc if the XML is malformed). It is also a cleaner way to test the XML value.